R/method-group.R

```diff
@@ -0,0 +1,40 @@
+group_generic_Math <- NULL
+group_generic_Ops <- NULL
+group_generic_Complex <- NULL
+group_generic_Sumary <- NULL
+
+on_load_define_group_generics <- function() {
+  group_generic_Math <<- new_generic("Math", "x")
+  group_generic_Ops <<- new_generic("Ops", c("e1", "e2"))
+  group_generic_Complex <<- new_generic("Complex", "z")
+  group_generic_Summary <<- new_generic("Summary", "x", function(x, ..., na.rm = FALSE) {
+    S7_dispatch()
+  })
+}
+
+#' @export
+Math.S7_object <- function(x, ...) {
+  group_generic_Math(x, ..., .Generic = .Generic)
+}
+
+#' @export
+Ops.S7_object <- function(e1, e2) {
```

Comment:
Need to move this back to `method-ops` since `Ops` is genuinely different to the other group generics because it's the only one where we want to override every specific generic with an S7 generic in order to get S7 multi-dispatch.